pkg/operator/hostetcdendpointcontroller/members.go

```diff
@@ -41,19 +43,31 @@ func (h *healthyEtcdMemberGetter) GetHealthyEtcdMembers() ([]string, error) {
 	}
 	hostnames := make([]string, 0)
 	for _, m := range member {
-		hostname := getEtcdName(m.PeerURLS)
+		hostname, err := getEtcdName(m.PeerURLS)
+		if err != nil {
+			return nil, err
+		}
 		hostnames = append(hostnames, hostname)
 	}
 	return hostnames, nil
 }
 
-func getEtcdName(peerURLs []string) string {
+func getEtcdName(peerURLs []string) (string, error) {
 	for _, peerURL := range peerURLs {
 		if strings.Contains(peerURL, "etcd-") {
-			return strings.TrimPrefix(strings.Split(peerURLs[0], ".")[0], "https://")
+			return strings.TrimPrefix(strings.Split(peerURLs[0], ".")[0], "https://"), nil
+		}
+		u, err := url.Parse(peerURL)
+		if err != nil {
+			return "", err
+		}
+		host, port, _ := net.SplitHostPort(u.Host)
+		//TODO peer port should be a global constant
+		if isIP(host) && port == "2380" {
+			return "etcd-bootstrap", nil
 		}
 	}
-	return ""
+	return "", nil
```

Comment:
is this an error? not found?

Reply:
yes, reaching that point is very unhealthy. I would think a lot of things are not going right for us to reach that state, it would be fair to panic at that point?

Reply:
put an error for now but we need to think about how we handle this.